src/datachain/query/dataset.py

```diff
@@ -1220,6 +1223,28 @@ def _order_columns(
     return [[d[n] for n in column_order] for d in column_dicts]
 
 
+def _remap_sys_id_dense(
+    columns: Iterable[ColumnElement], branch_offset: int
+) -> list[ColumnElement]:
+    """Return a new column list where sys__id is remapped to a dense row number.
+
+    The resulting sys__id is computed as:
+      row_number(order by sys__id) * 2 + branch_offset
+
+    This guarantees uniqueness across branches (even vs odd) and avoids
+    arithmetic overflow from scaling arbitrary pre-existing ids.
+    """
+    remapped: list[ColumnElement] = []
+    for col in columns:
+        if getattr(col, "name", None) == "sys__id":
+            rn = sqlalchemy.func.row_number().over(order_by=col)
```

Comment:
not sure about this one cc @dreadatour  - do we need reproducible order? or is it fine to drop it completely ... means each time we run we can a differnet sequence after union ... which is probably fine 🤔 ?

Reply:
I believe it is fine to ignore the order.

In ClickHouse order is not guaranteed and it was OK until now. But same time may be (?) someone rely on order as "undocumented feature" and this change might break this behavior.

Reply:
discussed with @dreadatour . No apparent reason to believe CH breaks on row_number(), we are waiting a response for a bit from CH team, but then we can merge this.